### ISSUE TITLE
Define help for just the root command 

### ISSUE THREAD

**Issue by kimskimchi** [spf13/cobra#1636]
I had a bit of difficulty adjusting the help output for the root command. 
To start off with the problem, I would like to group commands in `help` for only the root cmd, and keep rest of the command's`help` as the default. 

I am stuck for the following reasons:

1. I can use `SetHelpCommand` which does not get passed to children. However, I cannot overwrite the flag behaviour in my root's `Run` due to `execute` always returning ErrHelp: 
```
	if helpVal {
		return flag.ErrHelp
	}
```
ErrHelp states that "ErrHelp is the error returned if the flag -help is invoked but no such flag is defined.", but doesn't seem to be the case in this if statement. 

The end result here, is that 
a) it is localized to root for `root help` or `root`, but
b) when the flags are called `root -h` or `root --help`, it outputs a different format. 

Ideally, I'd like it to detect that the flag was defined by user, and skip to Run, instead of error'ing out. 

2. I can use `SetUsageTemplate` and use groups (using Annotations) to organize each one's subcommands. However, there seems to be no way to 'detect' which command is being run. This is the code I have so far 
```

func rootUsageTemplate(cmd *cobra.Command) string {
	var template = `Usage:{{if .Runnable}}
{{.UseLine}}{{end}}{{if .HasAvailableSubCommands}}
{{.CommandPath}} [command]{{end}}{{if gt (len .Aliases) 0}}

Aliases:
{{.NameAndAliases}}{{end}}{{if .HasExample}}
x
Examples:
{{.Example}}{{end}}{{if .HasAvailableSubCommands}}

` + groupCommands(cmd) + `

Flags:
{{.LocalFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}{{if .HasAvailableInheritedFlags}}

Global Flags:
{{.InheritedFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}{{if .HasHelpSubCommands}}

Additional help topics:{{range .Commands}}{{if .IsAdditionalHelpTopicCommand}}
{{rpad .CommandPath .CommandPathPadding}} {{.Short}}{{end}}{{end}}{{end}}{{if .HasAvailableSubCommands}}

Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
`

	return template
}
```
The entire template is the same as the default, with a small addition of "groupCommands" in the middle. I cannot figure out a way for this to detect the current command being called at, as it's not a `.groupCommands`. 

Basically, I want root cmd's help to be slightly modified, while keeping the rest the same. 

The first can be resolved by the if-statement being slightly modified - that if it was specified by user, it should just skip to `Run`, instead of always erroring out to ErrHelp. 
The second can be resolved if there is a function to accept UsageTemplate but turn off inheritance, OR to detect the command it's in.  

I am totally happy to PR either!

**Comment by jpmcb** [comment]
I'm not sure I follow.

Can you provide an example workflow of what you want vs what you've experienced? Thanks much!

**Comment by johnSchnake** [comment]
For the template you can add an extra template method to detect if its the root command: https://github.com/spf13/cobra/blob/eb3b6397b1b5d1b0a2cd66a9afe0520f480c0a87/cobra.go#L69

So use your custom template with some new method (e.g. "isRoot") and just write the template so that if that method returns true, to do something else.

You can grab the default help template and just modify it to take into account your 'if root...' logic so its the same otherwise.

**Comment by jkakar** [comment]
Thanks for sharing that context @johnSchnake! It helped me realize I could use `SetHelpTemplate` and pass in a template that uses the default unless `.HasParent` is `false`, in which case the template has a branch to render the custom help output shown for the top-level command only.
